config/config.go

```diff
@@ -21,34 +21,55 @@
 package config
 
 import (
-	"github.com/pkg/errors"
-	zanzibar "github.com/uber/zanzibar/runtime"
+	"os"
+
+	"github.com/uber/zanzibar/runtime"
 )
 
-// NewRuntimeConfigOrDie returns a static config struct that is pre-set with
-// the service configuration defaults
+// EnvConfig map from environment variable to config key and data type
+type EnvConfig map[string]struct {
+	Key      string `json:"key"`
+	DataType string `json:"dataType"`
+}
+
+// NewRuntimeConfigOrDie returns a static config struct
+// that is pre-set with the service configuration defaults
+// and overridden by service.env.config
 func NewRuntimeConfigOrDie(
 	files []string,
 	seedConfig map[string]interface{},
 ) *zanzibar.StaticConfig {
 	defaultConfig, err := defaultConfig()
 	if err != nil {
-		panic(errors.Wrap)
+		panic("error getting default config")
 	}
-	serviceConfig := make([]*zanzibar.ConfigOption, len(files)+1)
-	serviceConfig[0] = defaultConfig
-	for i, configFilePath := range files {
-		serviceConfig[i+1] = zanzibar.ConfigFilePath(configFilePath)
+
+	var serviceConfig []*zanzibar.ConfigOption
```

Comment:
Setting capacity to known files+1 length.  append() will not copy the array if there is capacity to append an item.